generator/k8s_client_template.go

```diff
@@ -34,14 +34,21 @@ func NewForConfig(cfg rest.Config) (Interface, error) {
 	if err != nil {
 		return nil, err
 	}
-	return NewFromControllerFactory(controllerFactory)
+	return NewFromControllerFactory(controllerFactory), nil
 }
 
-func NewFromControllerFactory(factory controller.SharedControllerFactory) (Interface, error) {
+func NewFromControllerFactory(factory controller.SharedControllerFactory) Interface {
 	return &Client{
 		controllerFactory: factory,
 		clientFactory:     factory.SharedCacheFactory().SharedClientFactory(),
-	}, nil
+	}
```

Comment:
This will cause a cascade of updates needed when we regenerate controllers in rancher https://github.com/rancher/rancher/blob/216bb7f5fc36e93ef5df2a47aaf10ce592e697f2/pkg/types/config/context.go#L138 but that's fine

Reply:
Yeah I figured since it's just rancher that'll be affected and this isn't doing anything for us it shouldn't matter. The changes to rancher to be compatible with this are here: https://github.com/rancher/rancher/pull/39876.

Reply:
I'm generally not a fan of changing public/exported methods like this, especially in projects like this one that I don't think we are planning to add major changes to in the future. It also doesn't appear to be related to the PR.

That being said, you are right that the code wasn't helping us, so I won't fight this one too much.